airflow/contrib/hooks/gcp_function_hook.py

```diff
@@ -0,0 +1,104 @@
+                 delegate_to=None):
+        super(GCFHook, self).__init__(gcp_conn_id, delegate_to)
+        self.api_version = api_version
+
+    def get_conn(self):
+        """Returns a Google Cloud Functions service object."""
+        if not self._conn:
+            http_authorized = self._authorize()
+            self._conn = build('cloudfunctions', self.api_version,
+                               http=http_authorized, cache_discovery=False)
+        return self._conn
+
+    def list_functions(self, location):
+        list_response = self.get_conn().projects().locations().functions().list(
+            parent=location).execute(num_retries=NUM_RETRIES)
+        return list_response.get("functions", [])
+
+    def create_new_function(self, location, body):
+        response = self.get_conn().projects().locations().functions().create(
+            location=location,
+            body=body
+        ).execute(num_retries=NUM_RETRIES)
+        operation_name = response["name"]
+        return self._wait_for_operation_to_complete(operation_name)
+
+    def update_function(self, name, body, update_mask):
+        response = self.get_conn().projects().locations().functions().patch(
+            updateMask=",".join(update_mask),
+            name=name,
+            body=body
+        ).execute(num_retries=NUM_RETRIES)
+        operation_name = response["name"]
+        return self._wait_for_operation_to_complete(operation_name)
+
+    def upload_function_zip(self, parent, zip_path):
+        response = self.get_conn().projects().locations().functions().generateUploadUrl(
+            parent=parent
+        ).execute(num_retries=NUM_RETRIES)
+        upload_url = response.get('uploadUrl')
+        with open(zip_path, 'rb') as fp:
+            requests.put(
+                url=upload_url,
+                data=fp.read(),
+                # Those two headers needs to be specified according to:
+                # https://cloud.google.com/functions/docs/reference/rest/v1/projects.locations.functions/generateUploadUrl
+                # nopep8
+                headers={
+                    'Content-type': 'application/zip',
+                    'x-goog-content-length-range': '0,104857600',
+                }
+            )
+        return upload_url
+
+    def delete_function(self, name):
+        response = self.get_conn().projects().locations().functions().delete(
+            name=name).execute(num_retries=NUM_RETRIES)
+        operation_name = response["name"]
+        return self._wait_for_operation_to_complete(operation_name)
+
+    def _wait_for_operation_to_complete(self, operation_name):
+        service = self.get_conn()
+        while True:
+            operation_response = service.operations().get(
+                name=operation_name,
+            ).execute(num_retries=NUM_RETRIES)
+            if operation_response.get("done"):
+                return operation_response["response"]
+            time.sleep(1)
```

Comment:
You may want to use a named constant

Reply:
Fixed in 4ad338b98739d65a7aad11d9090f5840b2624b3f